coordinator/internal/logic/auth/login.go

```diff
@@ -1,25 +1,104 @@
 package auth
 
 import (
+	"errors"
+	"fmt"
+
 	"github.com/gin-gonic/gin"
+	"github.com/scroll-tech/go-ethereum/log"
 	"gorm.io/gorm"
 
+	"scroll-tech/common/version"
+
+	"scroll-tech/coordinator/internal/config"
+	"scroll-tech/coordinator/internal/logic/verifier"
 	"scroll-tech/coordinator/internal/orm"
+	"scroll-tech/coordinator/internal/types"
 )
 
 // LoginLogic the auth logic
 type LoginLogic struct {
+	cfg          *config.Config
 	challengeOrm *orm.Challenge
+	chunkVks     map[string]struct{}
+	batchVKs     map[string]struct{}
+	bundleVks    map[string]struct{}
 }
 
 // NewLoginLogic new a LoginLogic
-func NewLoginLogic(db *gorm.DB) *LoginLogic {
-	return &LoginLogic{
+func NewLoginLogic(db *gorm.DB, cfg *config.Config, vf *verifier.Verifier) *LoginLogic {
+	l := &LoginLogic{
+		cfg:          cfg,
+		chunkVks:     make(map[string]struct{}),
+		batchVKs:     make(map[string]struct{}),
+		bundleVks:    make(map[string]struct{}),
 		challengeOrm: orm.NewChallenge(db),
 	}
+
+	for _, vk := range vf.ChunkVKMap {
+		l.chunkVks[vk] = struct{}{}
+	}
+
+	for _, vk := range vf.BatchVKMap {
+		l.batchVKs[vk] = struct{}{}
+	}
+
+	for _, vk := range vf.BundleVkMap {
+		l.bundleVks[vk] = struct{}{}
+	}
+
+	return l
 }
 
 // InsertChallengeString insert and check the challenge string is existed
 func (l *LoginLogic) InsertChallengeString(ctx *gin.Context, challenge string) error {
 	return l.challengeOrm.InsertChallenge(ctx.Copy(), challenge)
 }
+
+func (l *LoginLogic) Check(login *types.LoginParameter) error {
+	if login.PublicKey != "" {
```

Comment:
we can remove this check after darwin upgrade.

Reply:
yes